### PR TITLE
`Development`: Allow startup if no previous version in database exists

### PR DESCRIPTION
**Pull request by rstief**
# Can only be tested locally, no need to deploy!

#### General
- [x] This is a small issue that I tested locally and was confirmed by another developer ~on a test server~ locally.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/development-process.html#naming-conventions-for-github-pull-requests).

#### Server
- [x] **Important**: I implemented the changes with a very good performance and prevented too many (unnecessary) database calls.
- [x] I **strictly** followed the [server coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/).
- [x] I documented the Java code using JavaDoc style.

### Motivation and Context
Multiple people had problems when setting up Artemis for the first time. The problem occurs when the database is initialized but Artemis fails to start. This leads to the `artemis_version` table having no previous version. Then, all consecutive startups fail due to the missing version.

### Description
This PR allows Artemis to start even if the previous version is missing. (see also: [Slack](https://ls1tum.slack.com/archives/CQBB4PSMC/p1711629707784149))

**THIS COULD THEORETICALLY LEAD TO PROBLEMS WHEN PEOPLE WANT TO MIGRATE FROM A VERSION BETWEEN 5.10.2 and 5.12.8 TO THE NEWEST ARTEMIS VERSION**. We accept this risk because:
- These versions are now over one year old.
- It only affects people that last upgraded Artemis between September 2022 and December 2022 and now directly want to upgrade to the latest version.

### Steps for Testing
This has to be tested locally. Delete the entry from the `artemis_version` table. See that Artemis still starts.

#### Performance Review
- [ ] I (as a reviewer) confirm that the server changes (in particular related to database calls) are implemented with a very good performance
#### Code Review
- [x] Code Review 1
- [ ] Code Review 2
#### Manual Tests
- [x] Test 1
- [ ] Test 2

### Test Coverage
Unchanged

### CODE REVIEW

File: src/main/java/de/tum/in/www1/artemis/config/migration/DatabaseMigration.java [hunk 1]
```diff
@@ -171,9 +171,8 @@ private String getPreviousVersionElseThrow() {
             if (result.next()) {
                 return result.getString("latest_version");
             }
-            // if no version exists, we fail here
-            log.error(error);
-            System.exit(12);
+            // no version exists
```
Comment (Strohgelaender) [inline src/main/java/de/tum/in/www1/artemis/config/migration/DatabaseMigration.java:174]:
```suggestion
            // no version exists
            // we can proceed by creating the initial schema
```

Comment (rstief) [reply src/main/java/de/tum/in/www1/artemis/config/migration/DatabaseMigration.java:174]: Actually, it is more likely that the initial schema was already created, but Artemis failed to start completely due to other faulty config. This then leaves the `artemis_version` table empty.

However, I can adjust the comment to say something like "no version exists, so we continue"

File: src/main/java/de/tum/in/www1/artemis/config/migration/DatabaseMigration.java [hunk 2]
```diff
@@ -172,9 +172,8 @@ private String getPreviousVersionElseThrow() {
...
```
Comment (rstief) [inline src/main/java/de/tum/in/www1/artemis/config/migration/DatabaseMigration.java:175]:
```suggestion
            // if no version is recorded in the table, we proceed with the startup
```

**Comment by Strohgelaender** [comment]
Did you try to fix the issue addressed by https://github.com/ls1intum/Artemis/pull/8652 in this PR? Or should this be changed regardless of that bugfix?

**Review by JohannesWt — APPROVED** [review]
demonstrated in testing Session - works as expected 👍

**Review by coolchock — APPROVED** [review]
demonstrated in the testing session, the app starts as expected. Code LGTM

**Review by muradium — APPROVED** [review]
Presented in testing session, code LGTM as well

**Review by Hialus — APPROVED** [review]
Thanks for fixing this 🚀

**Review by krusche — APPROVED** [review]

**Pull request merged**